test/system/src/main/java/io/pravega/test/system/framework/services/PravegaProperties.java

```diff
@@ -0,0 +1,100 @@
+    private static final ImmutableMap<String, String> DEFAULT_PROPERTIES = ImmutableMap.<String, String>builder()
+            // Segment store properties
+            .put("autoScale.muteInSeconds", "120")
+            .put("autoScale.cooldownInSeconds", "120")
+            .put("autoScale.cacheExpiryInSeconds", "120")
+            .put("autoScale.cacheCleanUpInSeconds", "120")
+            .put("curator-default-session-timeout", "10000")
+            .put("bookkeeper.bkAckQuorumSize", "3")
+            // Controller properties.
+            .put("controller.transaction.maxLeaseValue", "60000")
+            .put("controller.retention.frequencyMinutes", "2")
+            .put("log.level", "DEBUG")
+            .build();
+
+    private final Map<String, String> properties = new HashMap<String, String>(DEFAULT_PROPERTIES);
+    private PravegaProperties(){}
+
+    /**
+     * This method adds a key only if it is absent, assuming the default values never need to change
+     *
+     * */
+    private void addProperty(String key, String value) {
+        properties.putIfAbsent(key, value);
+    }
+
+    public ImmutableMap<String, String> getProperties() {
+        return ImmutableMap.copyOf(properties);
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder strBuilder = new StringBuilder();
+        for (String e: this.getProperties().keySet()) {
+            strBuilder.append(e);
+            strBuilder.append(":");
+            strBuilder.append(this.getProperties().get(e));
+            strBuilder.append("|");
+        }
+        return strBuilder.toString();
+    }
+
+    /**
+     * This class overrides the lombok builder.
+     * Add all methods for building PravegaProperties (with/without Auth/TLS) to this class
+     */
+    public static final class PravegaPropertiesBuilder {
+        private static final PravegaProperties DEFAULT_INSTANCE = new PravegaProperties();
+
+        public static PravegaProperties buildWithSecurity() {
+            log.debug("Security Builder invoked.");
+            PravegaProperties ctx = new PravegaProperties();
+            addControllerProperties(ctx);
+            addSegmentStoreProperties(ctx);
+            return ctx;
+        }
+
+        public static PravegaProperties build() {
+            log.debug("Default Builder invoked.");
+            return DEFAULT_INSTANCE;
+        }
+
+        private static void addControllerProperties(PravegaProperties ctx) {
+            //auth
+            ctx.addProperty("controller.auth.enabled", "true");
+            ctx.addProperty("controller.auth.userPasswordFile", "/opt/pravega/conf/passwd");
+            ctx.addProperty("controller.auth.tokenSigningKey", "secret");
+        }
+
+
+        private static void addSegmentStoreProperties(PravegaProperties ctx) {
+            // auth
+            ctx.addProperty("autoScale.authEnabled", "true");
+            ctx.addProperty("autoScale.tokenSigningKey", "secret");
```

Comment:
should we make the `tokenSigningKey` configurable?

Reply:
Making configurable would help if we're going to execute system tests with different security setups.
Right now that is ruled out as all tests would just execute on 2 environments - 1 with all security enabled and 2 with security disabled.
Could make it configurable at a later point if we have the need to have multiple security environments.

Reply:
ok.